src/mixins/navigator/tagsProvider.js

```diff
@@ -45,53 +43,57 @@ export default {
       if (hideAvailableTags || selectedTags.length) return [];
       return availableTags;
     },
-    translatableTags: ({ availableTags }) => [HIDE_DEPRECATED, ...availableTags],
+    translatableTags: ({ availableTags }) => [FILTER_TAGS.hideDeprecated, ...availableTags],
   },
   methods: {
     extractTags(nodes, apiChanges) {
-      // extract tags from child nodes
-      const apiChangesTypesSet = new Set(Object.values(apiChanges));
-      const tagsSet = new Set(Object.values(FILTER_TAGS));
-      const generalTags = new Set([HIDE_DEPRECATED]);
-      // when API changes are available, remove the `HIDE_DEPRECATED` option
-      if (apiChangesTypesSet.size) {
-        generalTags.delete(HIDE_DEPRECATED);
-      }
-
+      const possibleTags = new Set(Object.values(FILTER_TAGS));
+      // create categories to order the availableTags
       const availableTags = {
         type: [],
         changes: [],
         other: [],
       };
+
+      // when API changes are available, remove the hide deprecated option
+      if (apiChanges && Object.values(apiChanges).length) {
+        possibleTags.delete(FILTER_TAGS.hideDeprecated);
+      }
+
       // iterate over the nodes to render
       for (const childID in nodes) {
         if (!Object.hasOwnProperty.call(nodes, childID)) {
           // eslint-disable-next-line no-continue
           continue;
         }
         // if there are no more tags to iterate over, end early
-        if (!tagsSet.size && !apiChangesTypesSet.size && !generalTags.size) {
+        if (!possibleTags.size) {
           break;
         }
         // extract props
         const { type, path, deprecated } = nodes[childID];
-        // grab the tagLabel
+
+        // add type tag
         const tag = TOPIC_TYPE_TO_TAG[type];
-        const changeType = apiChanges[path];
-        // try to match a tag
-        if (tagsSet.has(tag)) {
+        if (tag && possibleTags.has(tag)) {
```

Comment:
In theory, a tag should always be part of the possible tags because `TOPIC_TYPE_TO_TAG` is formed by `FILTER_TAGS` values. I think we should be safe on simplying to:

```suggestion
        if (tag) {
```

Reply:
I think checking if the tag exists in `possibleTags` is so that if we already found the tag and deleted it, we don't need to add it again. It's probably more straight forward to just check if the tag is already in `availableTags`, but it's harder to check there.

Reply:
I added a comment to make it clear.